File: ProblemSet/463. 整数排序/python/463 · 整数排序 插入排序.py
class Solution:
    """
    @param A: an integer array
    @return: nothing
    """
    def sortIntegers(self, A):
        for p in range(1, len(A)):
            tmp = A[p]
            i = p
            while i > 0 and A[i - 1] > tmp:
                A[i] = A[i - 1]
                i -= 1
            A[i] = tmp

File: ProblemSet/463. 整数排序/java/463 · 整数排序 插入排序.java
public class Solution {
    /**
     * @param A: an integer array
     * @return: nothing
     */
    public void sortIntegers(int[] A) {
        for (int p = 1; p < A.length; ++p) {
            int tmp = A[p];
            int i = p;
            for (; i > 0 && A[i - 1] > tmp; --i)
                A[i] = A[i - 1];
            A[i] = tmp;
        }
    }
}